packit/specfile.py

```diff
@@ -196,8 +196,21 @@ def add_patch(self, patch_metadata: PatchMetadata):
             logger.debug("There are no patches in the spec.")
             patch_number_offset = 0
 
+        if patch_metadata.patch_id is not None:
+            if patch_metadata.patch_id < patch_number_offset:
+                raise PackitException(
+                    f"Patch {patch_metadata.name} is suppose to have ID {patch_metadata.patch_id}"
+                    ". The spec file already contains patches and the latest one has "
+                    f"ID {patch_number_offset}. Packit is unable to fit this patch in the spec "
+                    "since it could break application in %prep. Please increase index of the "
+                    f"patch to {patch_number_offset + 1} at minimum."
```

Comment:
Maybe change to:

The 'patch_id' requested ({patch_metadata.patch_id}) for patch {patch_metadata.name} is less than or equal to the last used patch ID ({patch_number_offset}). Re-ordering the patches using the patch ID is not allowed.

Reply:
Your description is shorter and more clear - I also added another sentence to explain how to reorder commits:

    if you want to change the order of patches, please reorder commits